### PR TITLE
[TIR] Blockize keeping T.init in inner block when outer block does no reduction

### PR DESCRIPTION
**Pull request by MasterJH5574**
This PR enhances the behavior of tir.Schedule.blockize, so that when the outer block after blockization does no effective reduction, the `T.init` part will be kept in the inner block.

No new unit test is added as the update of the existing tests in this PR has already covered the correctness and ensured the expected behavior.

Co-authored-by: Bohan Hou <spectrometerh@gmail.com>

### CODE REVIEW

**Comment by MasterJH5574** [comment]
Depending on #15251.

File: tests/python/unittest/test_tir_schedule_blockize.py [hunk 1]
```diff
@@ -355,4 +353,5 @@ def after_blocks_blockize(
 
 
 if __name__ == "__main__":
-    tvm.testing.main()
+    test_blockize_outer()
```
Comment (yzh119) [inline tests/python/unittest/test_tir_schedule_blockize.py:356]: Keep it `tvm.testing.main()`.

**Review by yzh119 — APPROVED** [review]
LGTM

**Comment by MasterJH5574** [comment]
Closed since the intention of this PR seems to break MS tuning which is more complicated for fix.

**Pull request closed without merging**